src/storage/src/render/upsert.rs

```diff
@@ -602,7 +606,7 @@ mod tests {
     fn test_rehydrate_thin_multiple() {
         let mut packer = Row::default();
 
-        let key_indices = vec![2, 4];
+        let key_indices = vec![4, 2];
         let key = Row::pack([Datum::String("key1"), Datum::String("key2")]);
```

Comment:
This should now be `Row::pack([Datum::String("key2"), Datum::String("key1")]);`, but then the test will fail. So I think the logic in the PR is actually wrong

Reply:
The problem is that in `rehydrate`, you won't know where to put the key columns back into the values. You just insert them in order, which is wrong.

I guess this is why I made the `BTreeMap` in https://github.com/MaterializeInc/materialize/pull/12687 -- you do need to track both the key column index and the value column index for each of the thinned columns.